### PR TITLE
Improve CI vulnerabilities check - Closes #304

### LINKED ISSUE #304: Improve CI vulnerabilities check
We should have
- nightly builds to check if new vulnerabilities have been found for dependencies in `master`
- builds on PRs every time a change to `package.json` is detected to see if vulnerabilities would be introduced by the change

The settings should be public, but the results private. (This latter condition is not so important for Lisky in its current form, but will be important going forward, and insofar as this sets an example for adding these checks to other projects.)

### PR DESCRIPTION
**Pull request by ghost**
Run snyk only if package.json has changed since the nightly build.

### CODE REVIEW

File: Jenkinsfile.private [hunk 1]
```diff
@@ -0,0 +1,30 @@
+pipeline {
+	agent { node { label 'lisky' } }
+	stages {
+		stage('Prepare workspace') {
+			steps {
+				checkout scm
+			}
+		}
+		stage('Install dependencies') {
```
Comment (willclarktech) [inline Jenkinsfile.private]: Do we need this stage?

File: Jenkinsfile.private [hunk 2]
```diff
@@ -0,0 +1,30 @@
+pipeline {
+	agent { node { label 'lisky' } }
+	stages {
+		stage('Prepare workspace') {
+			steps {
+				checkout scm
```
Comment (willclarktech) [inline Jenkinsfile.private]: We don't want a `deleteDir()` because of the checksum? Will this cause us problems?

Comment (fchavant) [reply Jenkinsfile.private]: It was added a while back to workaround -- I assume -- tests not cleaning up after themselves; which is not relevant here.

File: Jenkinsfile.private (hunk 1 above)
Comment (fchavant) [reply Jenkinsfile.private]: Yes, snyk scans the `node_modules/` directory.

**Review by willclarktech — DISMISSED** [review]

**Review by willclarktech — DISMISSED** [review]

**Review by willclarktech — APPROVED** [review]

**Pull request merged**